bittensor/utils/subprocess/commit_reveal.py

```diff
@@ -0,0 +1,784 @@
+                    )
+    except Exception as e:
+        print(f"Error deleting expired commits: {e}")
+
+
+def revealed_commit(commit_hash: str):
+    """
+    Handles the revealed_hash command by updating the revealed status on the corresponding commit from the database using the commit hash.
+
+    Args:
+        commit_hash (str): The commit hash.
+    """
+    try:
+        with utils.DB(db_path=DB_PATH) as (conn, cursor):
+            sql = "SELECT COUNT(*) FROM commits WHERE commit_hash=?"
+            cursor.execute(sql, (commit_hash,))
+            count = cursor.fetchone()[0]
+            if count > 0:
+                # Update the revealed status in the database
+                update_sql = "UPDATE commits SET revealed = ? WHERE commit_hash = ?"
+                cursor.execute(update_sql, (True, commit_hash))
+                conn.commit()
+                print(
+                    f"\nUpdated revealed status on existing row with commit hash {commit_hash}"
+                )
+            else:
+                print(f"\nNo existing row found with commit hash {commit_hash}")
+    except Exception as e:
+        print(f"Error updating from table 'commits': {e}")
+
+
+def revealed_commit_batch(commit_hashes: List[str]):
+    """
+    Handles the revealed_batch_hash command by removing the corresponding commits from the database using the commit hashes.
+
+    Args:
+        commit_hashes (List[str]): The list of commit hashes.
+    """
+    try:
+        if not commit_hashes:
+            return
+        with utils.DB(db_path=DB_PATH) as (conn, cursor):
+            for commit_hash in commit_hashes:
+                sql = "SELECT COUNT(*) FROM commits WHERE commit_hash=?"
+                cursor.execute(sql, (commit_hash,))
+                count = cursor.fetchone()[0]
+                if count > 0:
+                    update_sql = "UPDATE commits SET revealed = ? WHERE commit_hash = ?"
+                    cursor.execute(update_sql, (True, commit_hash))
+                    conn.commit()
+                    print(
+                        f"\nUpdated revealed status on existing row with commit hash {commit_hash}"
+                    )
+                else:
+                    print(f"\nNo existing row found with commit hash {commit_hash}")
+    except Exception as e:
+        print(f"Error updating from table 'commits': {e}")
+
+
+def committed(commit: Commit):
+    """
+    Commits a new commit object to the database.
+
+    Args:
+        commit (Commit): The commit object to save.
+    """
+    with utils.DB(db_path=DB_PATH) as (conn, cursor):
+        commit_data = commit.to_dict()
+        column_names = ", ".join(commit_data.keys())
+        data = ", ".join(["?"] * len(commit_data))
+        sql = f"INSERT INTO commits ({column_names}) VALUES ({data})"
+        cursor.execute(sql, tuple(commit_data.values()))
+        conn.commit()
+    print(f"Committed data: {commit_data}")
+
+
+def get_all_commits() -> List[Commit]:
```

Comment:
if this function is used within module only, pls use `_get_all_commits` for name. The same for another functions.